### DISCUSSION TITLE
Running in Docker within LXC within Proxmox

### DISCUSSION THREAD

**Discussion by fergbrain** [haugene/docker-transmission-openvpn#2168] [Show and tell]
If you are running this container with Docker in an LXC on Proxmox and you get this error:
```
ERROR: Cannot open TUN/TAP dev /dev/net/tun: Operation not permitted (errno=1)
Exiting due to fatal error
```

You need to add the following to your `/etc/pve/lxc/ct_id.conf` file:

```
 lxc.cgroup2.devices.allow: c 10:200 rwm
 lxc.mount.entry: /dev/net dev/net none bind,create=dir
```

Sources/see also:
* https://forum.proxmox.com/threads/pve-7-openvpn-lxc-problem-cannot-open-tun-tap-dev.103081/
* https://pve.proxmox.com/wiki/OpenVPN_in_LXC

_N. B. I wanted to make sure this got captured someplace searchable, so I figured here was good. Let me know if there's a better place._

**Comment by pkishino** [comment]
I suggest editing the wiki and adding this ;) but great find

**Reply by fergbrain** [reply]
Forgive me, I don't see the wiki...where do I find it?

**Reply by pkishino** [reply]
http://haugene.github.io/docker-transmission-openvpn/

**Comment by EstebPain** [comment]
Thank you so much! this was driving me crazy after a Proxmox update
Each day a little less ignorant ñ_ñ

**Comment by mmol67** [comment]
Hello. I'm triying to get this container (and jsloan117/docker-openvpn-client that is based in this one) working into a LXD container. Followed this guide ttps://gist.github.com/wastrachan/8d75ecf39db1fc25ae73c93989a8df11 but keep getting this error:
`ERROR: Cannot open TUN/TAP dev /dev/net/tun: No such file or directory (errno=2)`
as part of the container log.
The only part I don know how to do is the LimitNPROC commentint... Could you help me?

**Reply by mmol67** [reply]
Fixed it adding the "device" clause in the YAML. I didn't need it outside the container, but I NEED it inside it
